src/main/daml/Daml/Finance/Instrument/Generic/Util.daml

```diff
@@ -62,6 +62,25 @@ createFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf coup
   prepareAndTagClaims couponClaims "Fix rate payment"
 -- FIXED_RATE_BOND_COUPON_CLAIMS_END
 
+-- | Calculate a fix rate amount (if a credit event has not yet happened) for each payment date and create claims.
+createConditionalCreditFixRatePaymentClaims : Applicative f => [SchedulePeriod] -> PeriodicSchedule -> Bool -> Decimal -> Decimal -> DayCountConventionEnum -> Deliverable -> Observable -> f [TaggedClaim]
+createConditionalCreditFixRatePaymentClaims schedule periodicSchedule useAdjustedDatesForDcf couponRate weightMultiplier dayCountConvention cashInstrumentCid defaultProbabilityReferenceId = do
+  let
+    creditEvent = Lte (Const 1.0, Observe defaultProbabilityReferenceId)
+    couponDatesAdjusted = map (.adjustedEndDate) schedule
+    couponAmounts = map (\p -> couponRate * (calcPeriodDcf dayCountConvention p useAdjustedDatesForDcf periodicSchedule.terminationDate periodicSchedule.frequency)) schedule
+    couponClaims = zipWith (\d a -> until creditEvent $ when (TimeGte $ d) $ scale (Const weightMultiplier * Const a) $ one cashInstrumentCid) couponDatesAdjusted couponAmounts
```

Comment:
I think you can just `and` the individual payments and put a single `until creditEvent` on top

Reply:
I changed it. Works fine!